src/Idris/Parser.idr

```diff
@@ -491,7 +491,7 @@ mutual
     <|> do start <- bounds (symbol "[")
            listExpr fname start indents
     <|> do b <- bounds (decoratedSymbol fname "!" *> simpleExpr fname indents)
-           pure (PBang (boundToFC fname b) b.val)
+           pure (PBang (virtualiseFC $ boundToFC fname b) b.val)
```

Comment:
I think we need to keep the semantics that `virtualiseFC` is only used on things
that do not exist in the source code. Same remark for the virtualise on `PCase`
and the various do blocks.

Reply:
I have had a look at `checkCase` and let's just go with virtualising the argument to
`PCase` because this is nasty...